src/date-util/date-util.ts

```diff
@@ -207,55 +211,96 @@ export namespace DateUtil {
     return parseByFormat(str, 'YYYYMMDDHHmmssSSS');
   }
 
-  export function format(d: Date, format: string): string {
-    const year = d.getFullYear();
-    const month = d.getMonth() + 1;
-    const day = d.getDate();
-    const hour = d.getHours();
-    const minute = d.getMinutes();
-    const second = d.getSeconds();
-    const millisecond = d.getMilliseconds();
+  export function setUTCOffset(d: DateType, offsetMinute: number): Date {
+    d = parse(d);
+
+    const year = d.getUTCFullYear();
+    const month = d.getUTCMonth();
+    const date = d.getUTCDate();
+    const hours = d.getUTCHours();
+    const minutes = d.getUTCMinutes();
+    const seconds = d.getUTCSeconds();
+    const milliSeconds = d.getUTCMilliseconds();
+
+    return new Date(
+      Date.UTC(
+        year,
+        month,
+        date,
+        hours,
+        minutes,
+        seconds,
+        milliSeconds + offsetMinute * ONE_SECOND * ONE_MINUTE_IN_SECOND
+      )
+    );
+  }
+
+  export function format(d: Date, opts?: DateFormatOpts): string {
+    // format의 기본 기준은 로컬 런타임으로 한다.
+    const isUTC = opts?.isUTC ?? false;
+    const format = opts?.format ? opts.format : isUTC ? DEFAULT_UTC_DATE_FORMAT : DEFAULT_LOCALE_DATE_FORMAT;
+
+    let dateInfo = {
+      year: d.getFullYear(),
+      month: d.getMonth() + 1,
+      date: d.getDate(),
+      hour: d.getHours(),
+      minute: d.getMinutes(),
+      second: d.getSeconds(),
+      millisecond: d.getMilliseconds(),
+    };
+    if (isUTC) {
+      dateInfo = {
+        year: d.getUTCFullYear(),
+        month: d.getUTCMonth() + 1,
+        date: d.getUTCDate(),
+        hour: d.getUTCHours(),
+        minute: d.getUTCMinutes(),
+        second: d.getUTCSeconds(),
+        millisecond: d.getUTCMilliseconds(),
+      };
+    }
 
     const FORMAT_RULE_REGEXP = /[yYmMdDhHsS]{1,4}/g;
 
     const formattedDate = format.replace(FORMAT_RULE_REGEXP, (match) => {
       switch (match) {
         case 'YYYY':
-          return `${year}`;
+          return `${dateInfo.year}`;
         case 'YY':
-          return `${year % 100}`;
+          return `${dateInfo.year}`.padStart(2, '0');
```

Comment:
`2022.padStart(2, '0') // 2022`
YY 면 22 가 나와야할 것 같은데 2022가 나오도록 되어 있습니다
```suggestion
          return String(dateInfo.year % 100).padStart(2, '0');
```
은 어떨까요?

YY 테스트 코드도 추가하면 좋을 것 같습니다!

Reply:
수정하고 테스트코드 추가했습니다!